cmd/eval.go

```diff
@@ -78,19 +88,26 @@ Example: combo eval -r REPLACE_ME=1,2,3 path/to/file
 				combination.WithSolveAhead(),
 			)
 
+			templateBuilder, err := template.NewBuilder(templateFile, combinations)
+			if err != nil {
+				return fmt.Errorf("failed to construct builder: %w", err)
+			}
+
 			ctx, cancel := context.WithCancel(context.Background())
 			defer cancel()
 
-			generatedFile, err := generator.Evaluate(ctx, string(file), combinations)
+			combinedTemplateDocuments, err := templateBuilder.Build(ctx)
 			if err != nil {
-				return fmt.Errorf("failed to generate combinations: %w", err)
+				return fmt.Errorf("failed to build documents with combinations: %w", err)
 			}
 
-			if err := validateFile([]byte(generatedFile)); err != nil {
-				return fmt.Errorf("failed to validate file generated: %w", err)
+			combinedTemplate := "---\n" + strings.Join(combinedTemplateDocuments, "\n---\n")
```

Comment:
[There is no standard convention for newlines across all operating systems](https://stackoverflow.com/questions/3720612/what-is-the-universal-newline-for-all-operating-systems-lf-and-cr), will this cause any issues when the binary is built/ran on different operating systems? We should probably specify which operator systems we support.

This comment does not apply for the combo controller as we control the operating system when building the image.

Reply:
This could probably be handled in the README. It would be a good idea to make an issue where we test the binary on various operating system and see how it handles them. It may also be possible to do that in CI to make sure we don't regress across platforms. Thoughts?

Reply:
SGTM.